### PR TITLE
Leader ideologies

### PR DESCRIPTION
**Pull request by TheUserWithNoName**
I am not even remotely done with leader ideologies but this the most I can get done at a core level before the next update.
Main focus is I added a bunch of Vanguardist ideologies and weighted them
Also assigned some of those ideologies to starting leaders and I assigned a few as historical characters

### CODE REVIEW

**Review by Arcadyi — APPROVED** [review]

File: common/ideologies/cwp_00_leader_authoritarian_ideologies.txt [hunk 1]
```diff
@@ -70,5 +70,334 @@ ideology_fascist = {
...
+					or = {
+						is_interest_group_type = ig_capitalists
+						is_interest_group_type = ig_bureaucrats
+					}
+				}
+			}
+			add = 50
+		}
+		else_if = {
+			limit = {
```
Comment (DZappala) [inline common/ideologies/cwp_00_leader_authoritarian_ideologies.txt:267]: THIS ADDITIONAL CHECK FOR LAWS NEEDS TO BE A SEPARATE IF. NOT ELSE IF.

Comment (DZappala) [reply common/ideologies/cwp_00_leader_authoritarian_ideologies.txt:267]: there are multiple of these in the file.

File: common/ideologies/cwp_00_leader_communist_ideologies.txt [hunk 2]
```diff
@@ -1,5 +1,5 @@
-﻿ideology_totalitarian = {
-	icon = "gfx/interface/icons/ideology_icons/ideology_leader/ideology_leader_ethno_nationalist.dds"
+﻿ideology_hardliner = {
+	icon = "gfx/interface/icons/ideology_icons/ideology_leader/ideology_leader_no_ideology.dds"
```
Comment (DZappala) [inline common/ideologies/cwp_00_leader_communist_ideologies.txt:2]: needs a different icon.

File: common/ideologies/cwp_00_leader_communist_ideologies.txt [hunk 3]
```diff
@@ -73,18 +75,28 @@
...
+			add = 100
+		}
+		
 	}
 }
 
-ideology_authoritarian = {
-	icon = "gfx/interface/icons/ideology_icons/ideology_leader/authoritarian.dds"
+ideology_revisionist = {
+	icon = "gfx/interface/icons/ideology_icons/ideology_leader/ideology_leader_no_ideology.dds"
```
Comment (DZappala) [inline common/ideologies/cwp_00_leader_communist_ideologies.txt:99]: different icon pls.

File: common/ideologies/cwp_00_leader_communist_ideologies.txt [hunk 4]
```diff
@@ -73,18 +75,28 @@
...
-						is_interest_group_type = ig_capitalists
-						is_interest_group_type = ig_entrepreneurs
+						is_interest_group_type = ig_laborists
+						is_interest_group_type = ig_ruralists
 					}
 				}
 			}
 			add = 50
 		}
+		else_if = {
```
Comment (DZappala) [inline common/ideologies/cwp_00_leader_communist_ideologies.txt:85]: shouldn't be an else if here.

File: common/ideologies/cwp_00_leader_communist_ideologies.txt [hunk 5]
```diff
@@ -32,39 +32,41 @@
...
+			nor = {
+				has_law = law_type:law_representative_democracy
+				has_law = law_type:law_consensus_democracy
+				has_law = law_type:law_direct_democracy
 			}
 		}
 	}
 	leader_weight = {
-		value = 100
+		value = 500
```
Comment (DZappala) [inline common/ideologies/cwp_00_leader_communist_ideologies.txt:57]: this weighting is insane.

File: common/ideologies/cwp_00_leader_communist_ideologies.txt [hunk 6]
```diff
@@ -95,125 +107,335 @@ ideology_authoritarian = {
...
```
Comment (DZappala) [inline common/ideologies/cwp_00_leader_communist_ideologies.txt:154]: why so high?

File: common/ideologies/cwp_00_leader_communist_ideologies.txt [hunk 7]
```diff
@@ -95,125 +107,335 @@ ideology_authoritarian = {
...
 						is_interest_group_type = ig_ruralists
-						is_interest_group_type = ig_elites
-						is_interest_group_type = ig_entrepreneurs
+						is_interest_group_type = ig_securocrats
 					}
 				}
 			}
+			add = 50
+		}
+		else_if = {
```
Comment (DZappala) [inline common/ideologies/cwp_00_leader_communist_ideologies.txt:185]: needs to be an if.

File: common/ideologies/cwp_00_leader_communist_ideologies.txt [hunk 8]
```diff
@@ -95,125 +107,335 @@ ideology_authoritarian = {
...
+					has_law = law_type:law_oligarchy
+				}
+			}
+			add = 100
+		}
+	}
+}
+#Will change to something better when if we remember the better name
+ideology_apparatchick = {
+	icon = "gfx/interface/icons/ideology_icons/ideology_leader/ideology_leader_no_ideology.dds"
```
Comment (DZappala) [inline common/ideologies/cwp_00_leader_communist_ideologies.txt:197]: diff icon.

File: common/ideologies/cwp_00_leader_communist_ideologies.txt [hunk 9]
```diff
@@ -95,125 +107,335 @@ ideology_authoritarian = {
...
+				has_law = law_type:law_council_elective
+			}
+			nor = {
+				has_law = law_type:law_representative_democracy
+				has_law = law_type:law_consensus_democracy
+				has_law = law_type:law_direct_democracy
+			}
+		}
+	}
+	leader_weight = {
```
Comment (DZappala) [inline common/ideologies/cwp_00_leader_communist_ideologies.txt:250]: lvl. 500 is still insane.

File: common/ideologies/cwp_00_leader_communist_ideologies.txt [hunk 10]
```diff
@@ -95,125 +107,335 @@ ideology_authoritarian = {
...
-					or = {
-						is_interest_group_type = ig_capitalists
-						is_interest_group_type = ig_bureaucrats
-					}
+					is_interest_group_type = ig_laborists
 				}
 			}
 			add = 50
 		}
+		else_if = {
```
Comment (DZappala) [inline common/ideologies/cwp_00_leader_communist_ideologies.txt:276]: if blocks.

File: common/ideologies/cwp_00_leader_communist_ideologies.txt [hunk 11]
```diff
@@ -95,125 +107,335 @@ ideology_authoritarian = {
...
+			add = 100
+		}
 	}
 }
 
-ideology_developmentalist = {
-	icon = "gfx/interface/icons/ideology_icons/ideology_leader/exp_ideology_leader_protectionism.dds"
+ideology_apparatchick = {
+	icon = "gfx/interface/icons/ideology_icons/ideology_leader/ideology_leader_no_ideology.dds"
 	character_ideology = yes
```
Comment (DZappala) [inline common/ideologies/cwp_00_leader_communist_ideologies.txt:289]: Duplicate of above??

File: common/ideologies/cwp_00_leader_communist_ideologies.txt [hunk 12]
```diff
@@ -95,125 +107,335 @@ ideology_authoritarian = {
...
+					has_law = law_type:law_oligarchy
+				}
+			}
+			add = 100
+		}
+	}
+}
+
+ideology_reformist = {
+	icon = "gfx/interface/icons/ideology_icons/ideology_leader/ideology_leader_no_ideology.dds"
```
Comment (DZappala) [inline common/ideologies/cwp_00_leader_communist_ideologies.txt:379]: stop using no ideology. it's confusing. pick a real icon.

**Review by DZappala — CHANGES_REQUESTED** [review]
Some broken logic and wild numbers need fixing. also icons, ya picked the totally wrong one to use as your placeholder, look at the icons and pick one that matches the vibe of the ideology, there's like 50 to pick from.

File: common/ideologies/cwp_00_leader_communist_ideologies.txt (hunk 12 above)
Comment (Mr-Jangoon) [reply common/ideologies/cwp_00_leader_communist_ideologies.txt:288]: oki

**Pull request merged**